### PR TITLE
Uds 1481

### PR DESCRIPTION
**Pull request by scott-williams-az**
### Links

- [Unity reference site](https://unity.web.asu.edu/)
- [JIRA ticket](https://asudev.jira.com/browse/UDS-1481)
- [Unity Design Kit](https://xd.adobe.com/view/56f6cb78-9af5-4b12-b4ce-ef319f71113f-03a5/)

### Checklist

- [x] Unity project successfully builds from root `yarn install` & `yarn build`
- [x] Commits do not contain multiple scopes
- [x] Add/updated examples
- [ ] Add/updated READMEs/docs
- [ ] No new console errors
- [ ] Accessibility checks

### Browsers

- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

### CODE REVIEW

**Review by mlsamuelson — COMMENTED** [review]
When https://github.com/ASU/asu-unity-stack/commit/930de5829824601058bb1ef4e3b413d8356cba4e went in, I missed the implications that we were making `components-core` a peer dependency for the `component-header` instead of the necessary bits being compiled into the package.

`component-header` is designed to be standalone from other packages in that it doesn't require `unity-bootstrap-theme`. Current usage of `components-core` within the header and in this PR is for the utilities `getCurrentScriptPath` and `idGenerator`, so we're not mis-using it, but it has the possibility of opening the door to that. For this one case does it make sense as a peerDep?

Possible paths forward: 
1. Keep `components-core` in `component-header` as is (peerDependency) but update README.md for the package so the installation instructions spell this out more clearly and convey how to use the package - could be as simple as pointing out the need to provide the peerDeps. (Also review other peerDependencies for the package - are there any we want to highlight and/or move to the dependencies?).
2. Revert back, for this one package, to pulling in the utility scripts by path, so we have the power and convenience of code-reuse, but not the dependency on `components-core` for users to manage.
3. Other ideas?

**Comment by mlsamuelson** [comment]
Shared functions folder is to be added in https://github.com/ASU/asu-unity-stack/pull/1160

**Comment by mlsamuelson** [comment]
Informed by https://github.com/ASU/asu-unity-stack/pull/1160#pullrequestreview-1704711882

File: packages/component-header/package.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
     "build-storybook": "build-storybook -s ./dist"
   },
   "peerDependencies": {
-    "@asu/components-core": "^2.1.0",
+    "@asu/components-core": "^2.1.0 || ^3.0.0",
```
Comment (mlsamuelson) [inline packages/component-header/package.json]: Remove `components-core` as peerDep. Move referenced code to `utilities/` and reference it by path so it gets compiled into package.

Also, let's remove peerDeps for `https://github.com/ASU/asu-unity-stack/pull/1160#pullrequestreview-1704711882` and `prop-types` and `styled-components`.

File: packages/component-footer/package.json [hunk 2]
```diff
@@ -36,8 +36,10 @@
     "storybook": "start-storybook -s ./dist -p 9070",
     "build-storybook": "build-storybook -s ./dist"
   },
+  "peerDependencies": {
+    "@asu/components-core": "^2.1.0 || ^3.0.0"
```
Comment (mlsamuelson) [inline packages/component-footer/package.json]: Maybe include this by path instead, too.

**Review by mlsamuelson — CHANGES_REQUESTED** [review]

**Comment by scott-williams-az** [comment]
I ended up moving more scripts than we originally planned, and I am not set on the folder named "internal" so we can change that if needed.

**Comment by mlsamuelson** [comment]
@scott-williams-az Let's change from `internal/` to `shared/`.

**Review by mlsamuelson — CHANGES_REQUESTED** [review]
Seeing some errors. Maybe the `sanitizeDangerousMarkup` export needs to be uncommented on https://github.com/ASU/asu-unity-stack/pull/1189/files#diff-d54e7a9426b0944ee79ed9740d313a0c066c5980e42b6ed50d29b1a71619b2a7R7

Screenshot 2023-11-16 at 4 09 24 PM

**Review by mlsamuelson — APPROVED** [review]

**Pull request merged**